api.go

```diff
@@ -80,6 +80,10 @@ type LineMatch struct {
 	LineEnd    int
 	LineNumber int
 
+	// LinesBefore and LinesAfter are only set when SearchOptions.NumContextLines is > 0
+	LinesBefore [][]byte
+	LinesAfter  [][]byte
```

Comment:
design: is it not easier to just have `Before []byte`? IE reading your code which does this extraction it would be much easier to just create a single `[]byte` rather than per line. Same goes for After